rest-api/config_api.py

```diff
@@ -64,45 +63,60 @@ def check_config_admin():
   raise Forbidden()
 
 
-class ConfigApi(base_api.BaseApi):
+class ConfigApi(Resource):
   """Api handlers for retrieving and setting config values."""
-
   method_decorators = [auth_required_config_admin]
 
-  def __init__(self):
-    super(ConfigApi, self).__init__(config.ConfigurationDAO())
-
-  def get_config_by_date(self, key, date):
-    result = config.get_config_that_was_active_at(key, api_util.parse_date(date))
-    return self.make_response_for_resource(self.dao.to_json(result))
-
-  def get(self, key=None):
+  def get(self, key=config.CONFIG_SINGLETON_KEY):
     date = request.args.get('date')
-    key = key or config.CONFIG_SINGLETON_KEY
-    if not date:
-      # Return the live config.
-      return super(ConfigApi, self).get(key)
-    else:
-      return self.get_config_by_date(key, date)
-
-  # Insert or update to configuration
-  def post(self, key=None):
-    key = key or config.CONFIG_SINGLETON_KEY
-    resource = request.get_json(force=True)
-    m = self.dao.from_json(resource, None, key)
-    self.validate_object(m, None)
-    self.dao.store(m)
-    return self.make_response_for_resource(self.dao.to_json(m))
+    if date is not None:
+      date = parse_date(date)
+    model = config.load(key, date=date)
+    data = model.configuration or {}
+    return self.make_response_for_resource(data)
 
-  def put(self, key=None):
-    return super(ConfigApi, self).put(key or config.CONFIG_SINGLETON_KEY)
-
-  def validate_object(self, obj, a_id=None):
-    super(ConfigApi, self).validate_object(obj, a_id)
-    if obj.key.id() != config.CONFIG_SINGLETON_KEY:
+  def post(self, key=config.CONFIG_SINGLETON_KEY):
+    resource = request.get_json(force=True)
+    keypath = [config.Configuration, key]
+    model = config.Configuration(key=ndb.Key(flat=keypath))
+    model.populate(configuration=resource)
+    self.validate(model)
+    config.store(model)
+    return self.make_response_for_resource(model.configuration)
+
+  def put(self, key=config.CONFIG_SINGLETON_KEY):
+    resource = request.get_json(force=True)
+    keypath = [config.Configuration, key]
+    model = config.Configuration(key=ndb.Key(flat=keypath))
+    model.populate(configuration=resource)
+
+    self.validate(model)
+    if not model.key.get():
+      raise NotFound('{} with key {} does not exist'.format('Configuration', model.key))
+
+    date = None
+    if config.getSettingJson(config.ALLOW_NONPROD_REQUESTS, False):
+      date = request.headers.get('x-pretend-date', None)
+      if date:
+        date = parse_date(date)
+
+    client_id = app_util.get_oauth_id()
+    config.store(model, date=date, client_id=client_id)
+    return self.make_response_for_resource(model.configuration)
+
+  def make_response_for_resource(self, data):
```

Comment:
You can delete this whole thing and just return model.configuration above. Configs don't have last_modified. (The purpose of this logic was for e-tag checking, but we don't do that on configs, either.)

Reply:
Nice! I wasn't sure if a stored config might have a last modified tag or not.